WeakAurasOptions/AceGUI-Widgets/AceGUIWidget-WeakAurasDisplayButton.lua

```diff
@@ -44,6 +48,221 @@ local function Show_Long_Tooltip(owner, description)
+                WeakAuras.Add(source.data)
+            end
+        else
+            error("Calling 'Group' with invalid source. Reload your UI to fix the display list.")
+        end
+    end,
+    -- remove source from its group or top-level list
+    ["Ungroup"] =  function(source)
+        if source and source.data.parent then
+            local parent = WeakAuras.GetData(source.data.parent)
+            local children = parent.controlledChildren
+            local index = source:GetGroupOrder()
+            if ensure(children, index, source.data.id) then
+                tremove(children, index)
+                source:SetGroup()
+                source.data.parent = nil
+                WeakAuras.Add(parent);
+                WeakAuras.UpdateGroupOrders(parent);
+                WeakAuras.ReloadGroupRegionOptions(parent);
+                WeakAuras.UpdateDisplayButton(parent);
+            else
+                error("Display thinks it is a member of a group which does not control it")
+            end
+        else
+            error("Calling 'Ungroup' with invalid source. Reload your UI to fix the display list.")
+        end
+    end,
+    -- move source inside its own group before or after target
+    ["Move"] = function(source, target, before )
+        if source and source.data.parent then
+            local parent = WeakAuras.GetData(source.data.parent)
+            local children = parent.controlledChildren
+            local i = source:GetGroupOrder()
+            if ensure(children, i, source.data.id) then
+                if target and target.data.parent then
+                    local j = target:GetGroupOrder()
+                    if ensure(children, j, target.data.id) then
+                        -- account for possible reorder
+                        j = i < j and j-1 or j
+                        -- account for insert position
+                        j = before and j or j+1
+                        tremove(children, i)
+                        tinsert(children, j, source.data.id)
+                    else
+                        error("Calling 'Move' with invalid target. Reload your UI to fix the display list.")
+                    end
+                else
+                    tremove(children, i)
+                    tinsert(children, 1, source.data.id)
+                end
+                WeakAuras.Add(parent)
+                WeakAuras.UpdateGroupOrders(parent)
+                WeakAuras.UpdateDisplayButton(parent)
+            else
+                error("Calling 'Move' with invalid source. Reload your UI to fix the display list.")
+            end
+        else
+            error("Calling 'Move' with invalid source. Reload your UI to fix the display list.")
+        end
+    end,
+}
+
+local Icons = {
+    ["Group"] = "Interface\\GossipFrame\\TrainerGossipIcon",
+    ["Ungroup"] = "Interface\\GossipFrame\\UnlearnGossipIcon",
+    ["Move"] = nil
+}
+
+local function GetAction(target, area, source)
```

Comment:
you haven't declared _source and _target at the top of the function it will be invalid